src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/GeneralPartnerController.java

```diff
@@ -127,4 +128,15 @@ public ResponseEntity<ValidationStatusResponse> getValidationStatus(@RequestAttr
 
         return ResponseEntity.ok().body(validationStatus);
     }
+
+    @GetMapping("/general-partners")
+    public ResponseEntity<List<GeneralPartnerDto>> getGeneralPartners(@RequestAttribute(TRANSACTION_KEY) Transaction transaction,
+                                                                      @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId) {
+        String transactionId = transaction.getId();
+        HashMap<String, Object> logMap = new HashMap<>();
+        logMap.put(URL_PARAM_TRANSACTION_ID, transactionId);
+        ApiLogger.infoContext(requestId, String.format("Retrieving list of general partners for transactionId %s", transactionId), logMap);
```

Comment:
Not sure you need the "for transactionId %s" in the log message here. The id is in the `logMap` so should already be output.